providers/pool.rb

```diff
@@ -120,120 +120,150 @@ def site_identifier
+    is_new_cpu_limit = is_new_value?(doc.root, "APPPOOL/add/cpu/@limit", @new_resource.cpu_limit.to_s)
+    is_new_cpu_smp_affinitized = is_new_value?(doc.root, "APPPOOL/add/cpu/@smpAffinitized", @new_resource.cpu_smp_affinitized.to_s)
+    is_new_cpu_reset_interval = is_new_value?(doc.root, "APPPOOL/add/cpu/@resetInterval", @new_resource.cpu_reset_interval.to_s) 
+    is_new_smp_processor_affinity_mask = is_new_value?(doc.root, "APPPOOL/add/cpu/@smpProcessorAffinityMask", @new_resource.smp_processor_affinity_mask.to_s) 
+    is_new_smp_processor_affinity_mask_2 = is_new_value?(doc.root, "APPPOOL/add/cpu/@smpProcessorAffinityMask2", @new_resource.smp_processor_affinity_mask_2.to_s) 
+
+    # Application Pool set commands
+    if ((is_new_auto_start || is_new_start_mode) or
+       (@new_resource.runtime_version && is_new_managed_runtime_version) or
+       (@new_resource.recycle_at_time && is_new_recycle_at_time) or
+       (@new_resource.recycle_after_time && is_new_recycle_after_time) or
+       (@new_resource.thirty_two_bit && is_new_enable_32_bit_app_on_win_64) or
+       (@new_resource.max_proc && is_new_max_processes) or
+       (@new_resource.queue_length && is_new_queue_length))
+      $was_updated = true
       cmd = "#{appcmd(node)} set apppool \"/apppool.name:#{@new_resource.pool_name}\""
-      cmd << " /managedRuntimeVersion:v#{@new_resource.runtime_version}"
-      Chef::Log.debug(cmd) if @new_resource.runtime_version
-      shell_out!(cmd)
-      @new_resource.updated_by_last_action(true)
-    end
-    if @new_resource.worker_idle_timeout && is_new_idle_timeout
-      was_updated = true
-      cmd = "#{appcmd(node)} set config /section:applicationPools"
-      cmd << " \"/[name='#{@new_resource.pool_name}'].processModel.idleTimeout:#{@new_resource.worker_idle_timeout}\""
+      cmd << " /autoStart:#{@new_resource.auto_start.to_s}" if is_new_auto_start
+      cmd << " /startMode:#{@new_resource.start_mode.to_s}" if is_new_start_mode
+      cmd << " /managedRuntimeVersion:v#{@new_resource.runtime_version}" if @new_resource.runtime_version && is_new_managed_runtime_version
+      cmd << " /recycling.periodicRestart.time:#{@new_resource.recycle_after_time}" if @new_resource.recycle_after_time && is_new_recycle_after_time
+      cmd << " /enable32BitAppOnWin64:#{@new_resource.thirty_two_bit}" if @new_resource.thirty_two_bit && is_new_enable_32_bit_app_on_win_64
+      cmd << " /-recycling.periodicRestart.schedule" if @new_resource.recycle_at_time && is_new_recycle_at_time
+      cmd << " /+recycling.periodicRestart.schedule.[value='#{@new_resource.recycle_at_time}']" if @new_resource.recycle_at_time && is_new_recycle_at_time
+      cmd << " /processModel.maxProcesses:#{@new_resource.max_proc}" if @new_resource.max_proc && is_new_max_processes
+      cmd << " /queueLength:#{@new_resource.queue_length}" if @new_resource.queue_length && is_new_queue_length
```

Comment:
Found recycle schedule times too, I think:

This will remove all schedules:
`appcmd set config /section:applicationPools /-[name='string'].recycling.periodicRestart.schedule`

This will add one:
`appcmd set config /section:applicationPools /+[name='string'].recycling.periodicRestart.schedule.[value='{value}']`